flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/RexNodeJsonSerdeTest.java

```diff
@@ -0,0 +1,831 @@
+
+    private static RexNode createFunctionCall(
+            SerdeContext serdeContext, ContextResolvedFunction resolvedFunction) {
+        final BridgingSqlFunction nonSerializableFunction =
+                BridgingSqlFunction.of(
+                        serdeContext.getFlinkContext(),
+                        serdeContext.getTypeFactory(),
+                        resolvedFunction);
+        return serdeContext
+                .getRexBuilder()
+                .makeCall(
+                        nonSerializableFunction,
+                        serdeContext
+                                .getRexBuilder()
+                                .makeLiteral(
+                                        12,
+                                        serdeContext
+                                                .getTypeFactory()
+                                                .createSqlType(SqlTypeName.INTEGER),
+                                        false));
+    }
+
+    private static SerdeContext serdeContext(
+            CatalogPlanCompilation planCompilationOption, CatalogPlanRestore planRestoreOption) {
+        final Configuration configuration = new Configuration();
+        configuration.set(TableConfigOptions.PLAN_RESTORE_CATALOG_OBJECTS, planRestoreOption);
+        configuration.set(TableConfigOptions.PLAN_COMPILE_CATALOG_OBJECTS, planCompilationOption);
+
+        return JsonSerdeTestUtil.configuredSerdeContext(configuration);
+    }
+
+    private static SerdeContext serdeContextWithPermanentFunction(
+            CatalogPlanCompilation planCompilationOption, CatalogPlanRestore planRestoreOption) {
+        final SerdeContext serdeContext = serdeContext(planCompilationOption, planRestoreOption);
+        serdeContext
+                .getFlinkContext()
+                .getFunctionCatalog()
+                .registerCatalogFunction(UNRESOLVED_FUNCTION_CAT_ID, SER_UDF_CLASS, false);
+        return serdeContext;
+    }
+
+    private static void dropPermanentFunction(SerdeContext serdeContext) {
+        serdeContext
+                .getFlinkContext()
+                .getFunctionCatalog()
+                .dropCatalogFunction(UNRESOLVED_FUNCTION_CAT_ID, false);
+    }
+
+    private static void modifyPermanentFunction(SerdeContext serdeContext) {
+        dropPermanentFunction(serdeContext);
+        serdeContext
+                .getFlinkContext()
+                .getFunctionCatalog()
+                .registerCatalogFunction(UNRESOLVED_FUNCTION_CAT_ID, SER_UDF_CLASS_OTHER, false);
+    }
+
+    private static void registerTemporaryFunction(SerdeContext serdeContext) {
+        serdeContext
+                .getFlinkContext()
+                .getFunctionCatalog()
+                .registerTemporaryCatalogFunction(
+                        UNRESOLVED_FUNCTION_CAT_ID, NON_SER_FUNCTION_DEF_IMPL, false);
+    }
+
+    private JsonNode serialize(
```

Comment:
Checking the calls, we are only testing permanent functions, or am I missing something?

Reply:
true, we can simply the calls